### PR TITLE
Problem details refactor

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@sammychinedu2ky  please review. Be sure to set hide whitespace
image
cc @tdykstra

File: fundamentals/middleware/problem-details-service/Program.cs [hunk 1]
```diff
@@ -104,12 +102,15 @@
...
+                                                           { } problemDetailsService)
         {
             (string Detail, string Type) details = mathErrorFeature.MathError switch
             {
-                MathErrorType.DivisionByZeroError => ("The number you inputed is zero", "https://en.wikipedia.org/wiki/Division_by_zero"),
-                _ => ("Negative or complex numbers are not handled", "https://en.wikipedia.org/wiki/Square_root")
+                MathErrorType.DivisionByZeroError => ("The number you inputed is zero",
+                "https://en.wikipedia.org/wiki/Division_by_zero"),
+                _ => ("Negative or complex numbers are not handled",
+                "https://en.wikipedia.org/wiki/Square_root")
```
Comment (Rick-Anderson) [inline fundamentals/middleware/problem-details-service/Program.cs:113]: I just split up a few lines so the code would show without a horizontal scroll bar on tablet sized devices.

Comment (sammychinedu2ky) [reply fundamentals/middleware/problem-details-service/Program.cs:113]: makes sense🤔

File: fundamentals/middleware/problem-details-service/Program.cs [hunk 2]
```diff
@@ -186,4 +183,70 @@ static bool HasPath(HttpContext context)
...
+app.MapControllers();
+
+app.Run();
+
+// Check if error message is defined for selected paths.
+static bool HasPath(HttpContext context)
+{
+    return true;
+}
+
```
Comment (Rick-Anderson) [inline fundamentals/middleware/problem-details-service/Program.cs:251]: @sammychinedu2ky this is going to get super messy.

Comment (Rick-Anderson) [reply fundamentals/middleware/problem-details-service/Program.cs:251]: @sammychinedu2ky  I added an API controller. Should I ask `@brunolins16` if there's a cleaner solution? Even with minimal API's, that's messy.

File: fundamentals/middleware/problem-details-service/Controllers/ValuesController.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+
+namespace ProblemDetailsWebApi.Controllers;
+
+[Route("api/[controller]/[action]")]
+[ApiController]
+public class ValuesController : ControllerBase
```
Comment (Rick-Anderson) [inline fundamentals/middleware/problem-details-service/Controllers/ValuesController.cs:8]: @sammychinedu2ky  PLEASE review controller. You can probably help clean it up.

Comment (sammychinedu2ky) [reply fundamentals/middleware/problem-details-service/Controllers/ValuesController.cs:8]: Ok sure let me look into it

File: fundamentals/middleware/problem-details-service/Program.cs (hunk 2 above)
Comment (sammychinedu2ky) [reply fundamentals/middleware/problem-details-service/Program.cs:251]: Yh I think that would be great. Would love to get his input and learn as well.

**Pull request merged**